### ISSUE TITLE
Free `vmThread->aotVMwithThreadInfo->_sharedCache` if AOT is disabled post-restore

### ISSUE THREAD

**Issue by dsouzai** [eclipse-openj9/openj9#16864]
When disabling AOT post-restore, `vmThread->aotVMwithThreadInfo->_sharedCache` should freed and set to NULL. However, it isn't obvious whether non compilation and java threads that could be running across a checkpoint/restore (because they aren't synchronized) could make use of it, in which case there could be a race. This issue tracks the work needed to free this object.

Implementing this safely depends on having https://github.com/eclipse-openj9/openj9/issues/16859 implemented.

**Comment by gacholio** [comment]
Seems to me this is somewhat irrelevant. The only reason to disable AOT in an image that was created with AOT is for debugging purposes, so saving memory doesn't really matter.

**Comment by dsouzai** [comment]
Yeah that's fair; I opened this mainly because I viewed as one of the loose ends left over from whatever we can get implemented for 0.38, but as you say, it may not really be worth doing.
